src/endpoints/proxy.rs

```diff
@@ -0,0 +1,125 @@
+use std::io::Cursor;
+
+use actix::ResponseFuture;
+use actix_web::{http::Method, HttpResponse, Path, State};
+use bytes::BytesMut;
+use failure::{Error, Fail};
+use futures::{future, Future, Stream};
+use itertools::Itertools;
+use symbolic::common::{CodeId, DebugId};
+use tokio::codec::{BytesCodec, FramedRead};
+
+use crate::actors::objects::{FetchObject, ObjectFileBytes, ObjectPurpose};
+use crate::app::{ServiceApp, ServiceState};
+use crate::types::{FileType, ObjectId, Scope};
+
+#[derive(Fail, Debug, Clone, Copy)]
+pub enum ProxyErrorKind {
+    #[fail(display = "failed to write object")]
+    Io,
+
+    #[fail(display = "failed sending message to objects actor")]
+    Mailbox,
+
+    #[fail(display = "failed to download object")]
+    Fetching,
+}
+
+symbolic::common::derive_failure!(
+    ProxyError,
+    ProxyErrorKind,
+    doc = "Errors happening while proxying to a symstore"
+);
+
+fn parse_symstore_path(path: &str) -> Option<(&'static [FileType], ObjectId)> {
+    let (leading_fn, signature, trailing_fn) = path.splitn(3, '/').collect_tuple()?;
```

Comment:
Do you want to use `splitn` here instead of `split`? I think we should assert that the path contains only three segments.

Reply:
We could, but since leading_fn needs to match trailing_fn that will be invalidated anyways.